lib/rubocop/cop/layout/empty_line_after_guard_clause.rb

```diff
@@ -79,7 +79,10 @@ def correct_style?(node)
         end
 
         def contains_guard_clause?(node)
-          node.if_branch&.guard_clause?
+          return unless (if_branch = node.if_branch)
+
+          if_branch.guard_clause? ||
+            if_branch.and_type? && if_branch.rhs.return_type?
```

Comment:
Do you think this should be part of the definition of the `#guard_clause?` method?

Reply:
This suggestion looks good to me 😻I updated this PR that in addition to `return`, keywords that break control, such as `break`, `next`, `raise`, and `fail` can also be detected.